File: ios/objNative/JSThemeType.m
//
//  JSThemeType.m
//  Supermap
//
//  Created by imobile-xzy on 2018/8/8.
//  Copyright © 2018年 Facebook. All rights reserved.
//

#import "JSThemeType.h"
#import "SuperMap/ThemeType.h"

@implementation JSThemeType
RCT_EXPORT_MODULE();
-(NSDictionary*)constantsToExport{
    return @{@"UNIQUE":@(TT_Unique),
             @"RANGE":@(TT_Range),
             @"GRAPH":@(TT_Graph),
             @"GRADUATEDSYMBOL":@(TT_GraduatedSymbol),
             @"DOTDENSITY":@(TT_DotDensity),
             @"LABEL":@(TT_label),
             @"CUSTOM":@(TT_Custom),
             @"GRIDUNIQUE":@(TT_GridUnique),
             @"GRIDRANGE":@(TT_GridRange),
           };
}

@end

File: android/src/main/java/com/supermap/rnsupermap/JSThemeType.java
package com.supermap.rnsupermap;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by Yangshanglong on 2018/8/7.
 */

public class JSThemeType extends ReactContextBaseJavaModule {
    public static final String REACT_CLASS="JSThemeType";

    public JSThemeType(ReactApplicationContext context){
        super(context);
    }

    @Override
    public String getName(){return REACT_CLASS;}

    @Override
    public Map<String, Object> getConstants() {
        final Map<String, Object> constants = new HashMap<>();
        constants.put("UNIQUE", 1);
        constants.put("RANGE", 2);
        constants.put("GRAPH", 3);
        constants.put("GRADUATEDSYMBOL", 4);
        constants.put("DOTDENSITY", 5);
        constants.put("LABEL", 7);
        constants.put("CUSTOM", 8);
        constants.put("GRIDUNIQUE", 11);
        constants.put("GRIDRANGE", 12);
        constants.put("LABELUNIQUE", 107);
        constants.put("LABELRANGE", 207);
        return constants;
    }
}